cpp/src/arrow/compute/kernels/CMakeLists.txt

```diff
@@ -18,11 +18,20 @@
 # ----------------------------------------------------------------------
 # Tests that don't require the full kernel library
 
+# Define arrow_compute_testing object library for common test files
```

Comment:
@kou Do these changes look ok? We were compiling `test_util.cc` 10 times...

Reply:
Yes. We can use `OBJECT` for this purpose.